app/assets/javascripts/models/beta/iteration.js

```diff
@@ -53,53 +53,93 @@ export const getIterationForStory = (story, project) => {
+  velocity - points > 0 ? velocity - points : 0;
+
+const createFillerSprints = (
+  quantityFillers,
+  initialNumber,
+  project,
+  velocity,
+  storyPoints
+) =>
+  _.times(quantityFillers, i => {
+    const remainingPoints = isLastFiller(i, quantityFillers) ? calcPointsOverflowedSprint(velocity, storyPoints) : 0;
     const sprintNumber = initialNumber + i;
-    const sprint = createSprint(
+    const emptyStories = [];
+    const sprintPoints = 0;
+    return createSprint(
       sprintNumber,
       getDateForIterationNumber(sprintNumber, project),
       true,
-      velocity
+      velocity,
+      emptyStories,
+      sprintPoints,
+      remainingPoints
     );
-    return sprint;
-  })
-};
+  });
+
+const createOverflowedSprint = (
+  velocity, 
+  storyPoints, 
+  fillersSprints, 
+  project,
+  overflowStory
+) => {
+  const numberSprint = getNextIterationNumber(fillersSprints);
+
+  return createSprint(
+    numberSprint,
+    getDateForIterationNumber(numberSprint, project),
+    true,
+    velocity,
+    [overflowStory],
+    storyPoints,
+    calcPointsOverflowedSprint(velocity, storyPoints)
+  );
+}
 
-const canTakeStory = (sprint, storyPoints) => {
-  if (!sprint.isFiller) {
-    return sprint.remainingPoints >= storyPoints;
-  }
-  return false;
-};
+export const canTakeStory = (sprint, storyPoints) => sprint.remainingPoints >= storyPoints;
 
-const calculateFillerSprintsQuantity = (storyPoints, velocity) => {
-  return Math.ceil((storyPoints - velocity) / velocity);
-};
+export const calculateFillerSprintsQuantity = (storyPoints, velocity) =>
+  Math.ceil((storyPoints - velocity) / velocity) || 0;
 
-const handleSprintsOverflow = (project, sprints, storyPoints, initialSprintNumber, velocity) => {
-  const overflow = calculateFillerSprintsQuantity(storyPoints, velocity);
-  const hasOverflown = overflow > 0;
-  const currentSprintNumber = sprints.length + initialSprintNumber;
-  if (hasOverflown) {
-    return [
-      ...sprints,
-      ...createFillerSprints(overflow, currentSprintNumber, project, velocity)
-    ];
-  }
-  return sprints;
+const addSprintWithOverflow = (project, sprints, story, velocity, firstSprintNumber) => {
+  const currentSprint = last(sprints);
+  const storyPoints = Story.getPoints(story);
+  const quantityFillers = calculateFillerSprintsQuantity(storyPoints, velocity);
+
+  if (isFirstSprint(currentSprint, firstSprintNumber) && haveNoPoints(currentSprint)) {
+    addStoryToSprint(project, sprints, 0, story, velocity);
+    return sprints;
+  } else {
+    const indexOverflow = calculateIndexSprint(currentSprint);
+    const fillerSprints = createFillerSprints(quantityFillers, indexOverflow, project, velocity, storyPoints);
+    const overflowedSprint = createOverflowedSprint(velocity, storyPoints, fillerSprints, project, story);
+    return [...sprints, ...fillerSprints, overflowedSprint];
+  } 
 };
 
+const calculateIndexSprint = currentSprint => currentSprint.number + 1;
```

Comment:
How about improving this name? something like `sprintIndexFor` would be more expressive
imagine like: `calculateIndexSprint(currentSprint)` vs `sprintIndexFor(currentSprint)` 
```suggestion
const sprintIndexFor = sprint => sprint.number + 1;
```

Reply:
:thinking: Sounds good... Gotcha!